app/src/main/java/org/stepic/droid/storage/DatabaseHelper.java

```diff
@@ -671,4 +683,28 @@ private void createCodeSubmissionTable(SQLiteDatabase db) {
                 + ")";
         db.execSQL(sql);
     }
+
+    private void createSearchQueryTable(SQLiteDatabase db) {
+        String sql = "CREATE TABLE " + DbStructureSearchQuery.SEARCH_QUERY
+                + " ("
+                + DbStructureSearchQuery.Column.QUERY_TEXT + WHITESPACE + TEXT_TYPE + " PRIMARY KEY, "
+                + DbStructureSearchQuery.Column.QUERY_TIMESTAMP + WHITESPACE + DATETIME_TYPE + WHITESPACE + DEFAULT + WHITESPACE + "(DATETIME('now', 'utc'))"
+                + ")";
+        db.execSQL(sql);
+    }
+
+    private void createSearchQueryTableSizeLimiterTrigger(SQLiteDatabase db) {
+        String sql = "CREATE TRIGGER IF NOT EXISTS" + WHITESPACE + DbStructureSearchQuery.LIMITER_TRIGGER_NAME + WHITESPACE
+                + "AFTER INSERT ON" + WHITESPACE + DbStructureSearchQuery.SEARCH_QUERY + WHITESPACE
+                + "BEGIN" + WHITESPACE
+                + "DELETE FROM" + WHITESPACE + DbStructureSearchQuery.SEARCH_QUERY + WHITESPACE
+                + "WHERE" + WHITESPACE + DbStructureSearchQuery.Column.QUERY_TEXT + WHITESPACE
+                + "NOT IN"
+                + "(SELECT" + WHITESPACE + DbStructureSearchQuery.Column.QUERY_TEXT + WHITESPACE
+                + "FROM" + WHITESPACE + DbStructureSearchQuery.SEARCH_QUERY + WHITESPACE
+                + "ORDER BY" + WHITESPACE + DbStructureSearchQuery.Column.QUERY_TIMESTAMP + WHITESPACE + "DESC" + WHITESPACE
+                + "LIMIT" + WHITESPACE + DbStructureSearchQuery.TABLE_SIZE_LIMIT + ");"
+                + "END;";
+        db.execSQL(sql);
```

Comment:
nice trigger for supporting only 100 rows, but on each insertion it will select TABLE_SIZE_LIMIT elements.